polkadot/node/core/pvf/common/src/pvf.rs

```diff
@@ -26,9 +26,9 @@ use std::{fmt, sync::Arc, time::Duration};
 /// Should be cheap to clone.
 #[derive(Clone, Encode, Decode)]
 pub struct PvfPrepData {
-	/// Wasm code (uncompressed)
-	code: Arc<Vec<u8>>,
-	/// Wasm code hash
+	/// Wasm code (maybe compressed)
```

Comment:
It's none of our business, `sp_maybe_compressed_blob` should decide if it has to be decompressed or returned as-is. But if needed, it can be detected by magic bytes in the beginning: https://github.com/paritytech/polkadot-sdk/blob/eb0a9e593fb6a0f2bbfdb75602a51f4923995529/substrate/primitives/maybe-compressed-blob/src/lib.rs#L30

Reply:
Okay, so it is encoded in the Vec<u8> :), I just thought it is something that we need to decide later down the line, but I see this is transparently handled for us by maybe-compressed-blob :)